### PR TITLE
Add support for modern watchOS/iOS architectures

### PR DESCRIPTION
**Pull request by CooperRS**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,67 +1,86 @@
 XCODE_DEVELOPER = $(shell xcode-select --print-path)
-IOS_PLATFORM ?= iPhoneOS
+PLATFORM ?= iPhoneOS
+MIN_VERSION ?= iphoneos-version-min=7.0
```
Comment (shagedorn) [inline Makefile:3]: Should we raise this to 9, or would that cause extra work?

File: Makefile [hunk 2]
```diff
@@ -1,67 +1,86 @@
...
-	${MAKE} arch ARCH=arm64 IOS_PLATFORM=iPhoneOS HOST=arm-apple-darwin
-	${MAKE} arch ARCH=i386 IOS_PLATFORM=iPhoneSimulator HOST=i386-apple-darwin
-	${MAKE} arch ARCH=x86_64 IOS_PLATFORM=iPhoneSimulator HOST=x86_64-apple-darwin
-
-PREFIX = ${CURDIR}/build/${ARCH}
+	${MAKE} arch ARCH=armv7k PLATFORM=watchOS HOST=arm-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=arm64_32 PLATFORM=watchOS HOST=arm-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=i386 PLATFORM=watchSimulator HOST=i386-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=x86_64 PLATFORM=watchSimulator HOST=x86_64-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=armv7 PLATFORM=iPhoneOS HOST=arm-apple-darwin MIN_VERSION=iphoneos-version-min=7.0
```
Comment (shagedorn) [inline Makefile:51]: Does this overwrite `MIN_VERSION` defined above? Can/should we raise it to 9?

File: Makefile [hunk 3]
```diff
@@ -1,67 +1,86 @@
...
 
-${LIBDIR}/libspatialite.a: ${LIBDIR}/libproj.a ${LIBDIR}/libgeos.a ${CURDIR}/spatialite
+${LIBDIR}/libspatialite.a: ${LIBDIR}/libproj.a ${LIBDIR}/libgeos.a ${LIBDIR}/libsqlite3.a ${CURDIR}/spatialite
 	cd spatialite && env \
 	CXX=${CXX} \
 	CC=${CC} \
 	CFLAGS="${CFLAGS} -Wno-error=implicit-function-declaration" \
 	CXXFLAGS="${CXXFLAGS} -Wno-error=implicit-function-declaration" \
-	LDFLAGS="${LDFLAGS} -liconv -lgeos -lgeos_c -lc++" ./configure --host=${HOST} --enable-freexl=no --enable-libxml2=no --prefix=${PREFIX} --with-geosconfig=${BINDIR}/geos-config --disable-shared && make clean install-strip
+	LDFLAGS="${LDFLAGS} -liconv -lgeos -lgeos_c -lc++" ./configure --host=${HOST} --enable-freexl=no --enable-libxml2=no --prefix=${PREFIX} --with-geosconfig=${BINDIR}/geos-config --disable-shared && make -j1 clean install-strip
```
Comment (shagedorn) [inline Makefile:77]: What's `-j1`? :D

File: patches/spatialite.patch [hunk 4]
```diff
@@ -6,10 +6,34 @@ diff -rupN spatialite/src/gaiaaux/gg_utf8.c spatialite-ios/src/gaiaaux/gg_utf8.c
  #if defined(__APPLE__) || defined(__ANDROID__)
```
Comment (shagedorn) [inline patches/spatialite.patch:6]: Is this patch applied automatically?

**Review by shagedorn — APPROVED** [review]
Mostly curiosity, up to you if we make extra changes or leave it as is.

File: Makefile (hunk 1 above)
Comment (CooperRS) [reply Makefile:3]: Hmm, I would vote for keeping it as low as supported by the `libspatialite`. If in doubt raising this only creates new warnings which we then need to fix.

File: Makefile (hunk 2 above)
Comment (CooperRS) [reply Makefile:51]: Yep, this overwrites the default above. For raising: See above.

File: Makefile (hunk 3 above)
Comment (CooperRS) [reply Makefile:77]: Well, passing `-j <number>` to `make` allows to build targets in parallel (`number` should be around 1.5x CPU Cores). However, it seems like `libspatialite` did not properly setup their `Makefile` such that building it with a value greater than 1 fails.

=> Just my little (poor) hack to allow me to build the dependencies in parallel while building `libspatialite` sequentially.

File: patches/spatialite.patch (hunk 4 above)
Comment (CooperRS) [reply patches/spatialite.patch:6]: Yep: https://github.com/iosphere/libspatialite-ios/blob/master-moers/update-spatialite#L3

**Comment by CooperRS** [comment]
I'll merge as soon as we have a working `libspatialite`

File: patches/spatialite.patch (hunk 4 above)
Comment (shagedorn) [reply patches/spatialite.patch:6]: Nice :)

**Comment by CooperRS** [comment]
I'll close this PR in favor of https://github.com/trailsapp/libspatialite-ios/pull/1

**Pull request closed without merging**